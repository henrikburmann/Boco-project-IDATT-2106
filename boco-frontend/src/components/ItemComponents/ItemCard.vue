<template>
  <!-- Item card, displays title, address, price per day and a picture -->
  <div class="mt-5 px-5">
    <div
      class="w-full h-full rounded bg-gray-100 ring-1 ring-gray-200 overflow-hidden display:inline-block correct-size cursor-pointer"
    >
      <div class="relative h-0 pb-[66.7%]">
        <img
          class="w-full h-full object-contain absolute"
          :src="item.img || require('../../assets/default-product.png')"
          alt="Item image"
        />
      </div>
      <div class="p-1 m-1 bottom-0">
        <p class="font-bold text-sm" id="title">{{ item.title }}</p>
        <p class="text-gray-700 text-xs" id="price">
          {{ item.pricePerDay }} kr
        </p>
        <p class="text-gray-700 text-xs font-bold" id="adress">
          {{ item.address }}
        </p>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    item: {
      img: String,
      address: String,
      title: String,
      pricePerDay: Number,
    },
  },
};
</script>
