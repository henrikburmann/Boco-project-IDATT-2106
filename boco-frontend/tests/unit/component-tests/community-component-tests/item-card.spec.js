import { mount } from "@vue/test-utils";
import ItemCard from "@/components/ItemComponents/ItemCard.vue";

describe("ItemCard component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ItemCard, {
      props: {
        item: {
          img: "String",
          address: "String",
          title: "String",
          pricePerDay: 0,
        },
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
