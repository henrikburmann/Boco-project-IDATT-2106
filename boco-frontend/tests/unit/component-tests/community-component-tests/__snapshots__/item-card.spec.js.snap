// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemCard component renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Item card, displays title, address, price per day and a picture -->
  <div
    class="mt-5 px-5"
  >
    <div
      class="w-full h-full rounded bg-gray-100 ring-1 ring-gray-200 overflow-hidden display:inline-block correct-size cursor-pointer"
    >
      <div
        class="relative h-0 pb-[66.7%]"
      >
        <img
          alt="Item image"
          class="w-full h-full object-contain absolute"
          src="String"
        />
      </div>
      <div
        class="p-1 m-1 bottom-0"
      >
        <p
          class="font-bold text-sm"
          id="title"
        >
          String
        </p>
        <p
          class="text-gray-700 text-xs"
          id="price"
        >
          0 kr 
        </p>
        <p
          class="text-gray-700 text-xs font-bold"
          id="adress"
        >
          String
        </p>
      </div>
    </div>
  </div>
  
</div>
`;
